### ISSUE TITLE
[Bug]: Highres fix freezes/crashes webui

### ISSUE THREAD

**Issue by ArchAngelAries** [lshqqytiger/stable-diffusion-webui-directml#265]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### What happened?

Updated to latest commit, runs normal on first generation, but second generation freezes/crashes at 50% progress when highres fix process is supposed to begin.

### Steps to reproduce the problem

1. update webui & extensions
2. generate 1st image with highres fix enabled
3. generate again and freeze/crash at 50%

### What should have happened?

normal continued generation

### Sysinfo

[sysinfo-2023-09-06-12-29.txt](https://github.com/lshqqytiger/stable-diffusion-webui-directml/files/12537315/sysinfo-2023-09-06-12-29.txt)

### What browsers do you use to access the UI ?

Microsoft Edge

### Console logs

```Shell
venv "D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\venv\Scripts\Python.exe"
fatal: No names found, cannot describe anything.
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: 1.6.0
Commit hash: 39e6d26f3b66805677cc3b5c2bc3bd624c347608
Faceswaplab : Use GPU requirements
Checking faceswaplab requirements
0.006981499987887219
Launching Web UI with arguments: --backend=directml --medvram --upcast-sampling --no-half-vae --precision full --no-half --disable-nan-check --use-cpu interrogate --opt-split-attention-v1 --autolaunch --api
no module 'xformers'. Processing without...
no module 'xformers'. Processing without...
No module 'xformers'. Proceeding without it.
Tag Autocomplete: Could not locate model-keyword extension, Lora trigger word completion will be limited to those added through the extra networks menu.
[-] ADetailer initialized. version: 23.9.1, num models: 11
2023-09-06 06:35:32,256 - ControlNet - INFO - ControlNet v1.1.407
ControlNet preprocessor location: D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-controlnet\annotator\downloads
2023-09-06 06:35:32,399 - ControlNet - INFO - ControlNet v1.1.407
Loading weights [8020cd0636] from D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\models\Stable-diffusion\00Anime_sebitamix_v10.safetensors
Creating model from config: D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\configs\v1-inference.yaml
Loading VAE weights specified in settings: D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\models\VAE\vae-ft-mse-840000-ema-pruned.safetensors
Applying attention optimization: Doggettx... done.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
Running on local URL:  http://127.0.0.1:7860
No Image data blocks found.

To create a public link, set `share=True` in `launch()`.
No Image data blocks found.
No Image data blocks found.
Startup time: 32.3s (prepare environment: 6.5s, import torch: 3.3s, import gradio: 1.4s, setup paths: 0.6s, initialize shared: 1.8s, other imports: 0.3s, list SD models: 0.3s, load scripts: 5.7s, create ui: 9.2s, gradio launch: 1.6s, add APIs: 0.2s, app_started_callback: 1.3s).
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
Restarting UI...
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
Closing server running on port: 7860
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
Tag Autocomplete: Could not locate model-keyword extension, Lora trigger word completion will be limited to those added through the extra networks menu.
[-] ADetailer initialized. version: 23.9.1, num models: 11
2023-09-06 06:36:50,530 - ControlNet - INFO - ControlNet v1.1.407
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-ar\scripts\sd-webui-ar.py:433: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  arc_calc_height = gr.Button(value="Calculate Height").style(
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-ar\scripts\sd-webui-ar.py:433: GradioDeprecationWarning: Use `scale` in place of full_width in the constructor. scale=1 will make the button expand, whereas 0 will not.
  arc_calc_height = gr.Button(value="Calculate Height").style(
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-ar\scripts\sd-webui-ar.py:441: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  arc_calc_width = gr.Button(value="Calculate Width").style(
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-ar\scripts\sd-webui-ar.py:441: GradioDeprecationWarning: Use `scale` in place of full_width in the constructor. scale=1 will make the button expand, whereas 0 will not.
  arc_calc_width = gr.Button(value="Calculate Width").style(
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\modules\gradio_extensons.py:25: GradioDeprecationWarning: `optional` parameter is deprecated, and it has no effect
  res = original_IOComponent_init(self, *args, **kwargs)
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-inpaint-anything\scripts\inpaint_anything.py:909: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  out_image = gr.Image(label="Inpainted image", elem_id="ia_out_image", type="pil",
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-inpaint-anything\scripts\inpaint_anything.py:925: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  cleaner_out_image = gr.Image(label="Cleaned image", elem_id="ia_cleaner_out_image", type="pil",
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-inpaint-anything\scripts\inpaint_anything.py:985: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  webui_out_image = gr.Image(label="Inpainted image", elem_id="ia_webui_out_image", type="pil",
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-inpaint-anything\scripts\inpaint_anything.py:1060: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  cn_out_image = gr.Image(label="Inpainted image", elem_id="ia_cn_out_image", type="pil",
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-inpaint-anything\scripts\inpaint_anything.py:1099: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  sam_image = gr.Image(label="Segment Anything image", elem_id="ia_sam_image", type="numpy", tool="sketch", brush_radius=8,
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-inpaint-anything\scripts\inpaint_anything.py:1110: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  sel_mask = gr.Image(label="Selected mask image", elem_id="ia_sel_mask", type="numpy", tool="sketch", brush_radius=12,
D:\AI-Stuff\AI\00_stable-diffusion-webui-directml\extensions\sd-webui-inpaint-anything\scripts\inpaint_anything.py:1113: GradioDeprecationWarning: The `style` method is deprecated. Please set these arguments in the constructor instead.
  with gr.Row().style(equal_height=False):
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
Running on local URL:  http://127.0.0.1:7860

To create a public link, set `share=True` in `launch()`.
Startup time: 28.3s (list SD models: 0.8s, load scripts: 13.7s, reload script modules: 0.5s, scripts list_optimizers: 0.5s, create ui: 11.5s, gradio launch: 0.2s, app_started_callback: 1.1s).
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
No Image data blocks found.
Model loaded in 98.8s (load weights from disk: 1.4s, create model: 0.3s, apply weights to model: 0.5s, apply float(): 0.6s, load VAE: 0.2s, load textual inversion embeddings: 95.6s, calculate empty prompt: 0.2s).
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:06<00:00,  2.91it/s]
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:14<00:00,  1.37it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 40/40 [00:26<00:00,  1.81it/s]
0: 640x480 1 face, 88.1ms
Speed: 3.0ms preprocess, 88.1ms inference, 1.0ms postprocess per image at shape (1, 3, 640, 480)
100%|████████████████████████████████████████████████████████████████████████████████████| 9/9 [00:02<00:00,  4.10it/s]
Total progress: 100%|██████████████████████████████████████████████████████████████████| 40/40 [00:33<00:00,  1.20it/s]
100%|██████████████████████████████████████████████████████████████████████████████████| 20/20 [00:04<00:00,  4.64it/s]
Total progress:  50%|█████████████████████████████████                                 | 20/40 [00:04<00:03,  5.63it/s]
```

**Comment by hookechoes** [comment]
As I commented in the thread about ultimate upscaler, I'm having the exact same problem you are.

Did a full reinstall even, no dice.

**Comment by rruki** [comment]
yep full reinstall doesn't work. 
if only we can downgrade to the previous version.
if anyone knows how to do that, it will be very much appreciated.

**Comment by ArchAngelAries** [comment]
> yep full reinstall doesn't work. if only we can downgrade to the previous version. if anyone knows how to do that, it will be very much appreciated.

I was able to successfully revert to an older commit that worked for me,  git checkout 6d881f9   ...... You can upgrade again once  the main branch is fixed by using git checkout master

**Comment by Jorolesvaldo** [comment]
> 

Hey, sorry, but im quite noobie at this. How can i get this older commit? I opened the link but didn't understand. Could you please help me?

**Comment by hookechoes** [comment]
> > 
> 
> Hey, sorry, but im quite noobie at this. How can i get this older commit? I opened the link but didn't understand. Could you please help me?

Right click your webui-user.bat file in your stable-diffusion-webui-directml folder and click edit.

Add this to an empty line:

git checkout 6d881f9

Should do the trick.

Thanks for the tip Archangel.

**Comment by Jorolesvaldo** [comment]
Well man, i actually tried it myself and juts typed "git checkout 6d881f9" and it worked, but thanks for all the help! I can finally use SD again!

**Comment by ArchAngelAries** [comment]
> > > 
> > 
> > 
> > Hey, sorry, but im quite noobie at this. How can i get this older commit? I opened the link but didn't understand. Could you please help me?
> 
> Right click your webui-user.bat file in your stable-diffusion-webui-directml folder and click edit.
> 
> Add this to an empty line:
> 
> git checkout [6d881f9](https://github.com/lshqqytiger/stable-diffusion-webui-directml/commit/6d881f974da7a2faf457f6ac64e409cb64edab28)
> 
> Should do the trick.
> 
> Thanks for the tip Archangel.

For sure! Happy to help with what I can. 

Oh, btw, you technically don't need to edit your bat file to do **git checkout**. 

1.) (If on Windows) Navigate to your SD Webui install folder

2) Click your explorer address bar and type "cmd" and hit enter

3) a python cmd window should launch

4) enter **git checkout #** (#= whatever commit hash you're trying to rollback to) and you've officially rolled back. 

Same method can be used to re-merge with the master commit by using **git checkout master** after launching cmd, aaaannnddd then you can also normally stay up to date with commits with this same method by using **git pull**.

**Comment by lshqqytiger** [comment]
May be fixed in 92849df26f73b416d396b95c3fb8c64070fe3ad8.

**Comment by ArchAngelAries** [comment]
> lshqqytiger

Can confirm, my issue has been resolved as of https://github.com/lshqqytiger/stable-diffusion-webui-directml/commit/92849df26f73b416d396b95c3fb8c64070fe3ad8. 
Thank you for all your hard work, lshqqytiger. You are so awesome! :)
